### PR TITLE
feat: add circular progress component

### PR DESCRIPTION
**Pull request by eljefe223**
# Description
Add circular progress to MSFT `<progress/>` component.

Closes: #1130 

Indeterminate:
Screen Shot 2019-07-22 at 2 25 12 PM
Screen Shot 2019-07-22 at 2 25 25 PM

Determinate:
Screen Shot 2019-07-22 at 2 25 47 PM
Screen Shot 2019-07-22 at 2 25 40 PM

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-class-name-contracts-msft/src/progress/index.ts [hunk 1]
```diff
@@ -4,16 +4,51 @@ import { ProgressClassNameContract as BaseProgressClassNameContract } from "@mic
  * The class name contract for the progress component
  */
 export interface ProgressClassNameContract extends BaseProgressClassNameContract {
+    /**
+     * The root of the circular progress component
+     */
+    progressCircular?: string;
```
Comment (nicholasrice) [inline packages/fast-components-class-name-contracts-msft/src/progress/index.ts]: I think circular would be a modifier on `progress` - `progress__circular`

File: packages/fast-components-class-name-contracts-msft/src/progress/index.ts [hunk 2]
```diff
@@ -4,16 +4,51 @@ import { ProgressClassNameContract as BaseProgressClassNameContract } from "@mic
...
+
     /**
      * The progress value indicator
      */
     progress_valueIndicator?: string;
 
+    /**
+     * The circular progress value indicator
+     */
+    progressCircular_valueIndicator?: string;
```
Comment (nicholasrice) [inline packages/fast-components-class-name-contracts-msft/src/progress/index.ts]: I don't think you need `progressCircular_valueIndicator` - you should just select for `progress__circular progress_valueIndicator`

File: packages/fast-components-class-name-contracts-msft/src/progress/index.ts [hunk 3]
```diff
@@ -4,16 +4,51 @@ import { ProgressClassNameContract as BaseProgressClassNameContract } from "@mic
...
+
     /**
      * The indeterminate progress indicator
      */
     progress_indicator?: string;
 
+    /**
+     * The indeterminate circular progress indicator
+     */
+    progressCircular_indicator?: string;
```
Comment (nicholasrice) [inline packages/fast-components-class-name-contracts-msft/src/progress/index.ts]: You wouldn't need this one - again, just select for `progress__circular progress_indicator`

File: packages/fast-components-class-name-contracts-msft/src/progress/index.ts [hunk 4]
```diff
@@ -4,16 +4,51 @@ import { ProgressClassNameContract as BaseProgressClassNameContract } from "@mic
...
  */
 export interface ProgressClassNameContract extends BaseProgressClassNameContract {
+    /**
+     * The root of the circular progress component
+     */
+    progressCircular?: string;
+
+    /**
+     * The progress size control modifier
+     */
```
Comment (nicholasrice) [inline packages/fast-components-class-name-contracts-msft/src/progress/index.ts]: I'm not following what these three classes are for by these descriptions - can you elaborate and align with the naming conventions I mentioned?

File: packages/fast-components-react-msft/src/progress/progress.tsx [hunk 5]
```diff
@@ -47,48 +48,134 @@ class Progress extends Foundation<ProgressHandledProps, ProgressUnhandledProps,
...
+                    `progress_dot__${index + 1}`
                 )}`;
 
                 return <span className={className} key={index} />;
             });
     }
+
+    private renderCircularBackground(): JSX.Element {
+        return (
+            <circle
```
Comment (nicholasrice) [inline packages/fast-components-react-msft/src/progress/progress.tsx]: This same circle is created three times - maybe abstract to a small functional component that can accept className and style?

File: packages/fast-components-react-msft/src/progress/progress.props.ts [hunk 6]
```diff
@@ -10,12 +10,29 @@ import {
...
+
 /* tslint:disable:no-empty-interface */
 export interface ProgressManagedClasses
     extends ManagedClasses<ProgressClassNameContract> {}
 export interface ProgressHandledProps
     extends Subtract<BaseProgressHandledProps, BaseProgressManagedClasses>,
-        ProgressManagedClasses {}
+        ProgressManagedClasses {
+    /**
+     * The progess is circular prop
```
Comment (janechu) [inline packages/fast-components-react-msft/src/progress/progress.props.ts]: nit: no need for "is"

File: packages/fast-components-react-msft/src/progress/progress.schema.ts [hunk 7]
```diff
@@ -18,6 +18,17 @@ export default {
             title: "HTML maxValue attribute",
             type: "number",
         },
+        circular: {
+            title: "Is circular",
```
Comment (janechu) [inline packages/fast-components-react-msft/src/progress/progress.schema.ts]: nit: no need for "Is"

File: packages/fast-components-class-name-contracts-msft/src/progress/index.ts (hunk 4 above)
Comment (janechu) [reply packages/fast-components-class-name-contracts-msft/src/progress/index.ts]: I'm also curious here, if `progress` is the root element and these are all modifiers on the root element I would expect `progress__circular`, `progress__control`, `progress__container`, `progress__page`. Your styling would then presumably be:

```
{
    progress__circular: {
        "&$progress__control": {
            // more specific stylings for circular control
        }
    }
}
```

File: packages/fast-components-react-msft/src/progress/progress.tsx [hunk 8]
```diff
@@ -47,48 +48,123 @@ class Progress extends Foundation<ProgressHandledProps, ProgressUnhandledProps,
...
+
+        return className;
+    }
+
     private progressIndicatorClasses(): string {
         return [
-            get(this.props, "managedClasses.progress_indicator"),
-            get(this.props, "managedClasses.progress_indicator__determinate"),
+            get(this.props.managedClasses, "progress_indicator"),
+            get(this.props.managedClasses, "progress_indicator__determinate"),
```
Comment (janechu) [inline packages/fast-components-react-msft/src/progress/progress.tsx]: nit: I'd prefer we refrain from updating these in the future, it's all personal preference and doesn't actually fix anything

File: packages/fast-components-class-name-contracts-msft/src/progress/index.ts (hunk 4 above)
Comment (eljefe223) [reply packages/fast-components-class-name-contracts-msft/src/progress/index.ts]: the size modifiers only apply to the circular progress.

File: packages/fast-components-react-msft/src/progress/progress.tsx (hunk 8 above)
Comment (eljefe223) [reply packages/fast-components-react-msft/src/progress/progress.tsx]: I'm just matching what was done elsewhere in the file

**Pull request merged**